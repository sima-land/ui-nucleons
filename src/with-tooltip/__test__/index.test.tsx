import { render, fireEvent } from '@testing-library/react';
import { WithTooltip } from '..';

describe('WithTooltip', () => {
  it('should works correctly', async () => {
    const { baseElement, findByTestId } = render(
      <WithTooltip tooltip='Hello, world!'>
        {(ref, toggle) => (
          <div data-testid='opener' ref={ref as any} onClick={() => toggle(true)}>
            Opener
          </div>
        )}
      </WithTooltip>,
    );

    expect(baseElement.querySelectorAll('[data-testid="tooltip"]')).toHaveLength(0);

    fireEvent.click(await findByTestId('opener'));

    expect(baseElement.querySelectorAll('[data-testid="tooltip"]')).toHaveLength(1);

    expect(baseElement).toMatchSnapshot();
  });

  it('should handle "onDismiss" prop', async () => {
    const spy = jest.fn();

    const { findByTestId } = render(
      <WithTooltip onDismiss={spy}>
        {(ref, toggle) => (
          <div data-testid='opener' ref={ref as any} onClick={() => toggle(true)}>
            Opener
          </div>
        )}
      </WithTooltip>,
    );

    expect(spy).toHaveBeenCalledTimes(0);

    fireEvent.click(await findByTestId('opener'));

    expect(spy).toHaveBeenCalledTimes(0);

    fireEvent.click(await findByTestId('tooltip:cross'));

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
