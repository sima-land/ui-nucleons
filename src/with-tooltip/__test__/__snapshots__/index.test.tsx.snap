// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithTooltip should works correctly 1`] = `
<body>
  <div>
    <div
      data-testid="opener"
    >
      Opener
    </div>
  </div>
  <div>
    <div
      class="root"
      data-testid="tooltip"
      style="position: absolute; top: 0px; left: 0px; opacity: 1; z-index: 1; max-height: -32px; transform: translate3d(16px, 16px, 0);"
    >
      <svg
        class="cross"
        data-testid="tooltip:cross"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.707 4.707a1 1 0 0 0-1.414-1.414L8 6.586 4.707 3.293a1 1 0 0 0-1.414 1.414L6.586 8l-3.293 3.293a1 1 0 1 0 1.414 1.414L8 9.414l3.293 3.293a1 1 0 0 0 1.414-1.414L9.414 8l3.293-3.293Z"
        />
      </svg>
      <div
        class="content"
      >
        Hello, world!
      </div>
    </div>
  </div>
</body>
`;
